src/System.Memory/tests/ArrayBufferWriter/ArrayBufferWriterTests.T.cs

```diff
@@ -0,0 +1,401 @@
+                WriteData(output, 100);
+                Assert.Throws<InvalidOperationException>(() => output.Advance(output.FreeCapacity + 1));
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SizeHints))]
+        public void GetSpan_DefaultCtor(int sizeHint)
+        {
+            {
+                var output = new ArrayBufferWriter<T>();
+                Span<T> span = output.GetSpan();
+                Assert.Equal(1, span.Length);
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>();
+                Span<T> span = output.GetSpan(sizeHint);
+                Assert.Equal(sizeHint == 0 ? 1 : sizeHint, span.Length);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SizeHints))]
+        public void GetSpan_InitSizeCtor(int sizeHint)
+        {
+            {
+                var output = new ArrayBufferWriter<T>(100);
+                Span<T> span = output.GetSpan();
+                Assert.Equal(100, span.Length);
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>(256);
+                Span<T> span = output.GetSpan(sizeHint);
+                Assert.Equal(sizeHint <= 256 ? 256 : sizeHint + 256, span.Length);
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>(1000);
+                Span<T> span = output.GetSpan(sizeHint);
+                Assert.Equal(sizeHint <= 1000 ? 1000 : sizeHint + 1000, span.Length);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SizeHints))]
+        public void GetMemory_DefaultCtor(int sizeHint)
+        {
+            {
+                var output = new ArrayBufferWriter<T>();
+                Memory<T> memory = output.GetMemory();
+                Assert.Equal(1, memory.Length);
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>();
+                Memory<T> memory = output.GetMemory(sizeHint);
+                Assert.Equal(sizeHint == 0 ? 1 : sizeHint, memory.Length);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(SizeHints))]
+        public void GetMemory_InitSizeCtor(int sizeHint)
+        {
+            {
+                var output = new ArrayBufferWriter<T>(100);
+                Memory<T> memory = output.GetMemory();
+                Assert.Equal(100, memory.Length);
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>(256);
+                Memory<T> memory = output.GetMemory(sizeHint);
+                Assert.Equal(sizeHint <= 256 ? 256 : sizeHint + 256, memory.Length);
+            }
+
+            {
+                var output = new ArrayBufferWriter<T>(1000);
+                Memory<T> memory = output.GetMemory(sizeHint);
+                Assert.Equal(sizeHint <= 1000 ? 1000 : sizeHint + 1000, memory.Length);
+            }
+        }
+
+        public bool IsX64 { get; } = IntPtr.Size >= 8;
```

Comment:
If it's greater than 8 that's not x64 :smile:

Reply:
Can `Environment.Is64BitProcess` or `Environment.Is64BitOperatingSystem` be used instead?